src/Language/PureScript/Label.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module Language.PureScript.Label (Label(..)) where
+
+import Prelude.Compat hiding (lex)
+import Data.Monoid ()
+import Data.String (IsString(..))
+import Data.Text (Text)
+import Text.Parsec.Combinator (eof)
+import qualified Data.Text as T
+import qualified Data.Aeson as A
+
+import Language.PureScript.Parser.Common (runTokenParser)
+import Language.PureScript.Parser.Lexer (TokenParser, lname, lex)
+import Language.PureScript.PSString (PSString, codePoints)
+
+-- |
+-- Labels are used as record keys and row entry names. Labels newtype PSString
+-- because records are indexable by PureScript strings at runtime.
+--
+newtype Label = Label PSString
+  deriving (Eq, Ord, IsString, Monoid, A.ToJSON, A.FromJSON)
+
+instance Show Label where
+  show (Label s) = either (const $ show s) (const cs) $ runParser lname $ T.pack cs
```

Comment:
I was trying to have it be suitable for pretty-printing PureScript code, like above. I'll change it to use `deriving` and export a separate function for pretty-printing.